### ISSUE TITLE
Cryptomator 2 Other Vaults

### ISSUE THREAD

**Issue by Quorum75** [cryptomator/ios#51]
When choosing **Open Existing Vault** / **Other File Provider** all choices are greyed out. 

 ![image](https://user-images.githubusercontent.com/49731083/124389995-f3c1ff00-dce9-11eb-94fb-6c6711a77958.png)

**Comment by tobihagemann** [comment]
This is actually not a bug and it's just how most of 3rd party file providers work. But I think we should make this very clear in the onboarding screen that this is expected.

**Comment by tobihagemann** [comment]
If anyone is wondering about the technical details: Apps need to support "picking folders" via `NSExtensionFileProviderSupportsPickingFolders` in the `Info.plist` of the file provider extension. However, this key doesn't seem to be officially documented anywhere by Apple. Twitter was the only [source](https://twitter.com/palmin/status/1177860144258076673) that we could find (actually the complete [reply chain](https://twitter.com/Dan2552/status/1177672714020167681) is hilarious).

We're not sure if we should support this key either because it doesn't seem to really work. But we're still investigating it during the beta. E.g., adding a vault inside a vault doesn't work (yet?).

**Comment by Quorum75** [comment]
I understand, Apple has never been very transparent on anything anyway. 

FYI it works with Boxcryptor for any Cloud services. However I don't know how they manage it.

**Comment by tobihagemann** [comment]
If I'm not mistaken, Boxcryptor doesn't have the option to add "other file providers" (via the `UIDocumentPickerViewController`). What you're seeing is that they support a lot of cloud services using their own implementations. Cryptomator actually supports a lot of "other" cloud serivces as well via WebDAV, see #37.

**Comment by Quorum75** [comment]
To be honest the process on macOS and iOS is a bit different. 
On macOS you can use Boxcryptor on any Could provider even if they are not on the list, like Mega for instance. 
On iOS you have a close list of Cloud services, however you can also use WebDAV to add whatever service you want. 

![image](https://user-images.githubusercontent.com/49731083/124454974-5a4a2a00-dd89-11eb-9076-3c32da603b88.png)

**Comment by ShredderWalnut** [comment]
I'm trying to access a Cryptomator vault from within Textastic but the Cryptomator extension is greyed out. Is this caused by the same issue in this thread? Will it be "fixed" in a update?

**Comment by phil1995** [comment]
Yes, you are right it is exactly the same reason (lack of `NSExtensionFileProviderSupportsPickingFolders` support) why Cryptomator is grayed out in Textastic and all other apps that need access to folders.
As described in https://github.com/cryptomator/ios/issues/48#issuecomment-898377602 we decided to not support `NSExtensionFileProviderSupportsPickingFolders` as it is undocumented and allows 3rd party apps to access the filesystem directly without interacting with our `FileProviderExtension`.

**Comment by toomaj** [comment]
> If I'm not mistaken, Boxcryptor doesn't have the option to add "other file providers" (via the `UIDocumentPickerViewController`). What you're seeing is that they support a lot of cloud services using their own implementations. Cryptomator actually supports a lot of "other" cloud serivces as well via WebDAV, see #37.

WebDav is not always the best choice as most cloud storage providers don't provide MFA/2FA on WebDav.

**Comment by emillienemann** [comment]
> If anyone is wondering about the technical details: Apps need to support "picking folders" via `NSExtensionFileProviderSupportsPickingFolders` in the `Info.plist` of the file provider extension. However, this key doesn't seem to be officially documented anywhere by Apple. Twitter was the only [source](https://twitter.com/palmin/status/1177860144258076673) that we could find (actually the complete [reply chain](https://twitter.com/Dan2552/status/1177672714020167681) is hilarious).
> 
> We're not sure if we should support this key either because it doesn't seem to really work. But we're still investigating it during the beta. E.g., adding a vault inside a vault doesn't work (yet?).

It'd be extremely useful to me. The only app I currently use that has this enabled is Working Copy and it's great for automation using shortcuts and many other things. It would also allow to people use Cryptomator folders in third party file browsers like FileBrowser, Documents or FE.
